src/ProjectOrigin.WalletSystem.Server/CommandHandlers/VerifySliceCommandHandler.cs

```diff
@@ -128,6 +132,13 @@ private async Task InsertIntoWallet(VerifySliceCommand receivedSlice, GranularCe
             await _unitOfWork.CertificateRepository.InsertCertificate(certificate);
         }
 
+        foreach (var hashedAttribute in receivedSlice.HashedAttributes)
+        {
+            if (certificateProjection.Attributes.Any(att => att.Key == hashedAttribute.Key
+                && att.Value == hashedAttribute.GetHashedValue()))
+                await _unitOfWork.CertificateRepository.InsertWalletAttribute(receivedSlice.WalletId, hashedAttribute);
```

Comment:
I found this hard to read. Maybe it is because of me just reviewing changes, but it took me some time to realize that `certificateProjection` is the certificate (projection) from the registry. For some reason I thought here that it was the Wallet's representation for a certificate

Reply:
refactored, hope it is cleaner now